Plugins/GitUIPluginInterfaces/IProcess.cs

```diff
@@ -47,18 +48,25 @@ public interface IProcess : IDisposable
         /// Blocks the calling thread until the process exits, or when this object is disposed.
         /// </summary>
         /// <returns>The process's exit code, or <c>null</c> if this object was disposed before the process exited.</returns>
-        int WaitForExit();
+        int WaitForProcessExit();
 
         /// <summary>
         /// Returns a task that completes when the process exits, or when this object is disposed.
         /// </summary>
         /// <returns>A task that yields the process's exit code, or <c>null</c> if this object was disposed before the process exited.</returns>
-        Task<int> WaitForExitAsync();
+        Task<int> WaitForProcessExitAsync();
+
+        /// <summary>
+        /// Instructs the process component to wait for the associated process to exit, or for the cancellationToken to be cancelled (cancells the process).
+        /// </summary>
+        /// <param name="token">An optional token to cancel the asynchronous operation.</param>
+        /// <returns>A task that will complete when the process has exited, cancellation has been requested, or an error occurs.</returns>
+        Task WaitForExitAsync(CancellationToken token);
```

Comment:
So now we have:
```cs
/// Blocks the calling thread until the process exits, or when this object is disposed.
int WaitForProcessExit();

/// Returns a task that completes when the process exits, or when this object is disposed.
Task<int> WaitForProcessExitAsync();

/// Instructs the process component to wait for the associated process to exit, or for the cancellationToken to be cancelled (cancells the process).
Task WaitForExitAsync(CancellationToken token);
```

Maybe we can do something like this, and remove the need for the new method?
```cs
            public Task<int> WaitForProcessExitAsync(CancellationToken token)
            {
                token.Register(() => _exitTaskCompletionSource.SetCanceled(token));
                return _exitTaskCompletionSource.Task;
            }
```

Reply:
Not really an option, exitTask is awaiten together with tasks for stdout and possibly stderr.

Reply:
I did not get this working, just renaming